### ISSUE TITLE
Output doesn't match C++ reference implementation in all cases

### ISSUE THREAD

**Issue by cimi** [LinusU/murmur128#1]
👋 Hi there!

I needed my JavaScript Murmurhash3 output to match the the C++ reference implementation and I noticed that this library produces different output in some cases:

| Input | C++ (reference) | murmur-128 |
| ---- | ---- | ---- |
| 'My hovercraft is full of eels.' | e3a186aee169ba6c6a8bd9343c68fa9c | 9acf132591e2e351c504b564a4897aa4 |
| 'My 🚀 is full of 🦎.' | e616d85ffee7f678dab461995b5bb90f | e616d85ffee7f678dab461995b5bb90f |
| '吉 星 高 照' | bb39ba50e4b2457c738e6d76b66d9304 | e3265bcaf6e98c729cafb605077e4b13 |

I  made an interactive comparison between different implementations, you can [try it here](https://cimi.io/murmurhash3js-revisited/). The page runs the C++ version through WebAssembly.

I’m passing the input in as utf-8 bytes:

```
murmur128x86(new TextEncoder().encode(str).buffer)
```

You can [see the source here](https://github.com/cimi/murmurhash3js-revisited/tree/docs).

Cheers!

**Comment by LinusU** [comment]
This is not good, the intention was absolutely to match the C++ implementation 👍 

Will take a look at this as soon as I can, PRs welcome!

**Comment by cimi** [comment]
Sorry, I won't have time to do this PR - I had already solved my problem by publishing [`murmurhash3js-revisited`](https://cimi.io/murmurhash3js-revisited/) when I found this one and added it to the comparison. 

One thing I noticed when looking at the source here is the conversion to bytes / array buffer before hashing:

```javascript
module.exports = function arrayBufferFromString (input) {
  var length = input.length
  var view = new Uint16Array(length)

  for (var i = 0; i < length; i++) {
    view[i] = input.charCodeAt(i)
  }

  return view.buffer
}
```

Using the `charCode` instead of a binary encoding doesn't seem right, although some strings with multibyte char codes do have matching results.

**Comment by LinusU** [comment]
Okay, so the problem might be that I'm using UTF-16 (or UCS-2) instead of UTF-8. This is documented in the readme here: https://github.com/LinusU/murmur-128#murmur128key-arraybuffer--string--arraybuffer

Might be better to switch to UTF-8 though 🤔

**Comment by LinusU** [comment]
Sorry for the super late reply on this. There was indeed a bug in my implementation as well, it's fixed with 72ee3abbd6332482931a651345845470bb87fabd and released in [🚢 0.1.2](https://github.com/LinusU/murmur-128/releases/tag/v0.1.2).

I've also switched from UTF-16/UCS-2 to UTF-8 here 14d4e518c7c163f6d4cc07bbc9cf466e46c31f0a. This is released as a breaking change in [🚢 0.2.2](https://github.com/LinusU/murmur-128/releases/tag/v0.2.0).

With this I believe that this can be closed, I'd appreciated if you could update the version in your comparison as well ☺️ 

If there is anything else we can do to cooperate/converge the libraries just let me know! 💪
